afew/MailMover.py

```diff
@@ -1,55 +1,58 @@
 # -*- coding: utf-8 -*-
 # SPDX-License-Identifier: ISC
 # Copyright (c) dtk <dtk@gmx.de>
+"""This module contain the MailMover class"""
 
-import notmuch
+import os
 import logging
-import os, shutil
+import shutil
+import sys
 from subprocess import check_call, CalledProcessError
-
-from .Database import Database
-from .utils import get_message_summary
 from datetime import date, datetime, timedelta
 import uuid
+import notmuch
+from .Database import Database
+from .utils import get_message_summary
 
 
 class MailMover(Database):
     '''
-    Move mail files matching a given notmuch query into a target maildir folder.
+    Move mail files matching a given notmuch query to a target maildir folder.
     '''
 
-
-    def __init__(self, max_age=0, rename = False, dry_run=False):
+    def __init__(self, max_age=0, rename=False, dry_run=False):
         super(MailMover, self).__init__()
         self.db = notmuch.Database(self.db_path)
         self.query = 'folder:{folder} AND {subquery}'
         if max_age:
             days = timedelta(int(max_age))
             start = date.today() - days
             now = datetime.now()
-            self.query += ' AND {start}..{now}'.format(start=start.strftime('%s'),
-                                                       now=now.strftime('%s'))
+            self.query += ' AND {start}..{now}'.format(
+                start=start.strftime('%s'),
+                now=now.strftime('%s'))
         self.dry_run = dry_run
         self.rename = rename
 
+    """ return the new name """
     def get_new_name(self, fname, destination):
         if self.rename:
             return os.path.join(
-                            destination,
-                            # construct a new filename, composed of a made-up ID and the flags part
-                            # of the original filename.
-                            str(uuid.uuid1()) + ':' + os.path.basename(fname).split(':')[-1]
-                        )
+                destination,
+                # construct a new filename, composed of a made-up ID
+                # and the flags part of the original filename.
+                filename = '{}:{}'.format(uuid.uuid1(),
+                          os.path.basename(fname).split(':')[-1]))
         else:
-            return destination
+            return os.path.join(destination, filename)
 
+    """ Move mails in folder maildir according to the given rules. """
     def move(self, maildir, rules):
         '''
         Move mails in folder maildir according to the given rules.
         '''
         # identify and move messages
         logging.info("checking mails in '{}'".format(maildir))
-        to_delete_fnames = []
         moved = False
         for query in rules.keys():
```

Comment:
Could you assure that the mailbox dirs do exist. I would like to only change the config, without having to additionally create a new directory manually:

```
    for ncs in 'tmp new cur'.split(): 
        destination = os.path.join(self.db_path,rules[query],ncs)
        os.makedirs(destination, mode=0o700, exist_ok=True)
```

Reply:
On that point I am not sure that behavior should be default,
to my pov, the mailmover should move from existing mailbox to an other (existing).

I am not sure, this case may append often, but if there is a real use case behind this,
we could easily add this check, feel free to add more details @rpuntaie maybe I am missing something.